File: files/300-399/325/CH-325.py
import pandas as pd

path = "300-399/325/CH-325 Date Calculation.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=7)
test = pd.read_excel(path, usecols="D:D", skiprows=1, nrows=7)\
    .assign(**{"End Time": lambda df: pd.to_datetime(df["End Time"], dayfirst=True)})

input['Start Date'] = pd.to_datetime(input['Start Date'], dayfirst=True)
input['Duration [h:m:s]'] = pd.to_timedelta(input['Duration [h:m:s]'])
input['End Time Calc'] = input['Start Date'] + input['Duration [h:m:s]']

print(input['End Time Calc'] == test['End Time']) # One result incorect

File: files/300-399/325/CH-325.R
library(tidyverse)
library(readxl)

path = "300-399/325/CH-325 Date Calculation.xlsx"
input = read_excel(path, range = "B2:C8", col_types = c("text", "numeric"))
test  = read_excel(path, range = "D2:D8") %>%
  mutate(`End Time` = dmy_hms(`End Time`))

result = input %>%
  mutate(`Start Date` = dmy_hms(`Start Date`),
         `End Time` = `Start Date` + ddays(`Duration [h:m:s]`))

result$`End Time` == test$`End Time`
# first date incorrect